File: LawyerSys/Data/ScaffoldedModels/Judicial_Document.cs
using System;
using System.Collections.Generic;

namespace LawyerSys.Data.ScaffoldedModels;

public partial class Judicial_Document
{
    public int Id { get; set; }

    public string Doc_Type { get; set; } = null!;

    public int Doc_Num { get; set; }

    public string Doc_Details { get; set; } = null!;

    public string Notes { get; set; } = null!;

    public int Num_Of_Agent { get; set; }

    public int Customers_Id { get; set; }

    public virtual Customer Customers { get; set; } = null!;
}

File: MobileApp/lib/features/judicial/models/judicial_document.dart
class JudicialDocument {

  JudicialDocument({
    required this.id,
    required this.docType,
    required this.docNum,
    required this.docDetails,
    required this.notes,
    required this.numOfAgent,
    required this.customerId,
    this.customerName,
  });

  factory JudicialDocument.fromJson(Map<String, dynamic> json) => JudicialDocument(
        id: json['id'] as int? ?? 0,
        docType: json['docType']?.toString() ?? '',
        docNum: json['docNum'] as int? ?? 0,
        docDetails: json['docDetails']?.toString() ?? '',
        notes: json['notes']?.toString() ?? '',
        numOfAgent: json['numOfAgent'] as int? ?? 0,
        customerId: json['customerId'] as int? ?? 0,
        customerName: json['customerName']?.toString(),
      );
  final int id;
  final String docType;
  final int docNum;
  final String docDetails;
  final String notes;
  final int numOfAgent;
  final int customerId;
  final String? customerName;

  Map<String, dynamic> toJson() => {
        'id': id,
        'docType': docType,
        'docNum': docNum,
        'docDetails': docDetails,
        'notes': notes,
        'numOfAgent': numOfAgent,
        'customerId': customerId,
        'customerName': customerName,
      };
}
